### PR TITLE
coroot: init at 1.5.8

### PR DESCRIPTION
**Pull request by errnoh**
Adding package for [coroot](https://github.com/coroot/coroot), the server part of Coroot monitoring tool. I'll add the client version in a separate PR.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/co/coroot/package.nix
Comment (zimward) [inline pkgs/by-name/co/coroot/package.nix:18]:
```suggestion
    repo = "coroot";
```

**Review by zimward — COMMENTED** [review]
for consistency with `owner`

File: pkgs/by-name/co/coroot/package.nix
Comment (zimward) [inline pkgs/by-name/co/coroot/package.nix:51]:
```suggestion
  meta = {
    description = "Open-source APM & Observability tool";
    homepage = "https://coroot.com";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ errnoh ];
  };
  See: #208242
```

**Review by zimward — COMMENTED** [review]
other than the `meta` it does look good to me, but i know too little about the `npm` stuff to make a final verdict.

File: pkgs/by-name/co/coroot/package.nix
Comment (errnoh) [reply pkgs/by-name/co/coroot/package.nix]: fixed in latest commit :+1:

**Comment by mattpolzin** [comment]
I got an error building this on `aarch64-darwin` that appears to be described by:
```
Building subPackage .
# github.com/grafana/pyroscope-go/godeltaprof/internal/pprof
vendor/github.com/grafana/pyroscope-go/godeltaprof/internal/pprof/delta_mutex.go:30:20: undefined: runtime_cyclesPerSecond
vendor/github.com/grafana/pyroscope-go/godeltaprof/internal/pprof/proto.go:320:8: undefined: runtime_expandFinalInlineFrame
```

**Comment by SomeoneSerge** [comment]
@ofborg build

**Comment by SomeoneSerge** [comment]
@ofborg build coroot

**Comment by errnoh** [comment]
@mattpolzin good find! Turns out that Go version had bumped to 1.23 and that had broken a dependency. ( https://github.com/grafana/pyroscope-go/issues/117 )

I'll submit a PR to them as well, but for now I added a patch to fix the issue for us. (also bumped version to 1.5.8 while at it)

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 346844` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coroot</li>
  </ul>
</details>

File: pkgs/by-name/co/coroot/package.nix
Comment (mattpolzin) [inline pkgs/by-name/co/coroot/package.nix:53]:
```suggestion
    maintainers = with lib.maintainers; [ errnoh ];
    mainProgram = "coroot";
```

Comment (errnoh) [reply pkgs/by-name/co/coroot/package.nix:53]: fixed!

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 346844` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coroot</li>
  </ul>
</details>

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 346844` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coroot</li>
  </ul>
</details>

**Review by mattpolzin — APPROVED** [review]
Ideally patches are fetched (e.g. from open PRs or unreleased commits) but since you've stated that the fix is forthcoming instead of pre-existing, I think this PR is looking good.

**Comment by errnoh** [comment]
@mattpolzin no that's completely valid feedback! I did in fact make a PR to upstream shortly after posting that message ( https://github.com/coroot/coroot/pull/357 ). I've now changed this package to use that patch instead.

This does mean that I'll need you to press that approval button once again :sweat_smile:

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 346844` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coroot</li>
  </ul>
</details>

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 346844` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coroot</li>
  </ul>
</details>

**Review by mattpolzin — APPROVED** [review]

**Comment by mattpolzin** [comment]
I actually think this suffers from a different problem that I was forgetting about temporarily -- until this PR merges, it's possible for the patch to change (for the PR to get pushed to) so it isn't a stable way to refer to the patch. If the PR merges quickly then no harm anyway, otherwise I may have steered you wrong by suggesting a PR (specifically an unmerged one) was a good time to fetch the patch remotely.

**Comment by errnoh** [comment]
@mattpolzin I did consider if that would be a problem, but seemed like a minor one since:
- PR by me so it's unlikely that someone else would add commits to it
- contents of the PR are fully tool generated and don't contain any human input (`go get` + `go mod tidy`)
- if the PR changes the hash change prevents it from doing anything malicious other than breaking future builds

But that said, the concern is still valid and there's a way around it. Instead of referring to the PR diff we can refer to the commit diff. That way it's both idiomatic from nixpkgs perspective as it avoids adding unnecessary patches to the repo, as well as stable as it wont change even if new commits happen in the PR.

So this is what I changed in the latest version :)

**Comment by mattpolzin** [comment]
I do very much appreciate your follow up here. Technically the issue of stability is a bit hairier than described in your last comment, though. It’s common to allow maintainers to modify PRs so that’s one way a PR might get changed by another person (even potentially force-pushed which would destroy the previous commit). Without checking whether you’ve allowed that on this current PR or not, there’s actually even another way that a commit could get blown away which is via a squash and merge. Given that your current PR is only one commit, I’m not recalling at the moment if a squash and merge would modify the hash of that commit or recognize there’s no point in doing so. 

All that just to state the ways a commit might not be stable until merged into the main branch in general (possible they don’t both apply in this case). 

Since the risk of the commit changing is low and I have been able to find precedent for merging a patch-pull just like this one (prior to acceptance upstream), I’m _not_ asking you to change this PR again 😄

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 346844` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coroot</li>
  </ul>
</details>

**Review by mattpolzin — APPROVED** [review]

**Pull request merged**
